FlowCal/io.py

```diff
@@ -627,27 +673,35 @@ def __init__(self, infile):
         if self._header.analysis_begin and self._header.analysis_end:
             # Prioritize ANALYSIS segment offsets specified in HEADER over
             # offsets specified in TEXT segment.
-            self._analysis = read_fcs_text_segment(
-                buf=f,
-                begin=self._header.analysis_begin,
-                end=self._header.analysis_end,
-                delim=delim)[0]
+            try:
+                self._analysis = read_fcs_text_segment(
+                    buf=f,
+                    begin=self._header.analysis_begin,
+                    end=self._header.analysis_end,
+                    delim=delim)[0]
+            except:
+                warnings.warn("ignoring ANALYSIS segment")
+                self._analysis = {}
         elif self._header.version in ('FCS3.0', 'FCS3.1'):
             analysis_begin = int(self._text['$BEGINANALYSIS'])
             analysis_end = int(self._text['$ENDANALYSIS'])
             if analysis_begin and analysis_end:
-                self._analysis = read_fcs_text_segment(
-                    buf=f,
-                    begin=analysis_begin,
-                    end=analysis_end,
-                    delim=delim)[0]
+                try:
+                    self._analysis = read_fcs_text_segment(
+                        buf=f,
+                        begin=analysis_begin,
+                        end=analysis_end,
+                        delim=delim)[0]
+                except:
+                    warnings.warn("ignoring ANALYSIS segment")
+                    self._analysis = {}
             else:
                 self._analysis = {}
         else:
             self._analysis = {}
         
         # Import DATA segment
-        param_ranges = [int(self._text['$P{0}R'.format(p)])
+        param_ranges = [float(self._text['$P{0}R'.format(p)])
```

Comment:
Ahh ok. It appears that only the `Time` channel reports a floating point `$PnR`, which makes a little more sense.

`$PnR` values constitute an optional parameter to `fc.io.read_fcs_data_segment()` which are only used if `$DATATYPE` is `I` to mask off unused bits (which I've never actually seen change the data, but strictly adheres to the standard). The `$DATATYPE` of this file is `F` (single precision floating point), so the `$PnR` values are never used by `fc.io.read_fcs_data_segment()`. If `$DATATYPE` were `I`, then we'd probably hit an error (how would you even interpret a floating point `$PnR` value, though? This is what I was originally getting hung up on).

Furthermore, I agree with @castillohair that the standard doesn't expressly forbid floating point values for `$PnR` parameters, so the most robust approach to accommodating this wrinkle is to report `param_ranges` as a `float`. Assuming sensible `$PnR` values are still reported when `$DATATYPE` is `I`, they should still serve their purpose (masking off unused bits) as `float`s.